projects/observability/src/shared/dashboard/data/graphql/explorer-visualization/explorer-visualization-cartesian-data-source.model.ts

```diff
@@ -46,17 +43,8 @@ export class ExplorerVisualizationCartesianDataSourceModel extends ExploreCartes
     };
   }
 
-  protected buildRequestState(interval: TimeDuration | 'AUTO'): ExploreRequestState | undefined {
-    if (this.request?.series.length === 0 || this.request?.context === undefined) {
-      return undefined;
-    }
-
-    return {
-      series: this.request.series,
-      context: this.request.context,
-      interval: interval,
-      groupBy: this.request.groupBy,
-      groupByLimit: this.request.groupByLimit
-    };
+  protected buildRequestState(): ExploreRequestState | undefined {
+    // Unused since fetchResults is overriden, but abstract so requires a def
```

Comment:
This one's a bit weird. This is an abstract method used in the parent's fetchResults which we override in this class. It's not really relevant to this implementation since fetchResults uses an observable of state rather than building its own.

This was actually breaking the explorer, (although no one noticed) since the previous implementation was mapping the response data based on the initial state, not the current one.

Reply:
Yeah. AFAIR, the `ExploreVisualizationRequest` did not have all the latest request data. I think we emit on `exploreQuery$: Observable<TimeUnaware<GraphQlExploreRequest>>;` with latest interval data (possibly more). When i tested it last, using `ExploreVisualizationRequest` directly did break few use cases for me. So for correctness, I kept it the same way for this data source (but fixed the dependency in the parent data source).

Reply:
If this is working, then you don't need to override `fetchResults`. Just `return this.request` inside `buildRequestState`

Reply:
As a user interacts with the explorer, all of the fields on `ExploreVisualizationRequest` may change, and the new queries are emitted on that observable - so we should never really be reading from `ExploreVisualizationRequest` directly (it's basically like the angular router snapshot vs observable state, but with more confusing naming), but we also can't build state from it either - we should always use the latest from the observable. One way to see the current issue is going to the explorer, selecting a group by key and removing the interval. Instead of getting a segmented aggregation, nothing renders (because we process the result thinking it still has an interval).

Reply:
ah okay. It will still be confusing to a new developer. Do you think we should change the design a little bit (as a tech debt)?

Reply:
yeah, for sure.